### ISSUE TITLE
Animation

### ISSUE THREAD

**Issue by codbellamy** [codbellamy/Purple-Guy#7]
Write code to enable animation. Since we are using vsync, we know that each time the entity is updated, another frame has passed. We can increment an animation timer. Then, we can use a getter to determine the animation frame requested. We can then display the image based on which animation frame the entity is currently attempting to display. We can check which direction the entity is going by using some trigonometry on the velocity vector. A less math-intensive way of doing this is by managing flags when the entity is moving. This would require setting the flag each time velocity is added to the entity and adding to any function that currently manipulates either the position or the velocity of the entity.

In summary:
Easier method (trig) but more processor intensive.
Harder method (flags) but faster.

**Comment by codbellamy** [comment]
The next step should be to work towards giving animations to entities. The current challenge will be how to handle the case when an Entity might not have any animations and does not require an instance of an animation manager.
